core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -339,35 +339,26 @@ object NextTurnAutomation {
         )
     }
 
-    private fun chooseBeliefs(civInfo: CivilizationInfo, beliefContainer: BeliefContainer): HashSet<Belief> {
+    private fun chooseBeliefs(civInfo: CivilizationInfo, beliefsToChoose: Counter<BeliefType>): HashSet<Belief> {
```

Comment:
Is that the first use of `Counter<T>` where `T != String::class`? Wow! Congrats!

Reply:
Yeah that turned out to be much less useful than I expected

Reply:
On the other hand, Counter serializes horribly (using explicit class annotations for trivial Int's), and I always wanted to try how that would look if we un-generic'ed that class. Made some attempts at hooking into the serializer to teach it better manners, testing with precisely such a Counter<enum>, but got blocked by enum as an itself generic class type not being available at compile time to feed the setSerializer thingy, so I went with <*> and then couldn't access needed methods from the superclass due to some creepy "you're not allowed to call super because you're on Java 6", obviously untrue...